runelite-client/src/main/java/net/runelite/client/plugins/hunter/HunterPlugin.java

```diff
@@ -377,6 +391,79 @@ else if (containsBoulder) // For traps like deadfalls. This is different because
 		lastTickLocalPlayerLocation = client.getLocalPlayer().getWorldLocation();
 	}
 
+	@Subscribe
+	public void onInteractingChanged(InteractingChanged interactingChanged)
+	{
+		Actor target = interactingChanged.getTarget();
+		Actor source = interactingChanged.getSource();
+
+		mapActorToChinTarget(target, source);
+	}
+
+	/**
+	 *	 				DEAD TRAP MECHANICS
+	 *	 When spawning a trap, an invisible null NPC is spawned at the same location.
+	 *	 This "null" npc handles all the interactions between a trap and a target.
+	 * 	 If two hunter traps target the same Actor, then the trap
+	 * 	 which did not catch the target will get stuck in a "dead" state.
+	 * 	 It will no longer be able to re-target another actor.
+	 * 	 Therefore, the user can remove these traps and re-place them.
+	 *
+	 *	 The code below keeps track of which traps interacted with every
+	 *	 chinchompa. If any traps are left over in this list when
+	 *	 the chinchompa is caught, the trap enters a dead state.
+	 */
+	private void mapActorToChinTarget(Actor target, Actor source)
+	{
+		if (target != null && source != null)
+		{
+			// May be a trap (trap source name is null)
+			if (source.getName() == null)
```

Comment:
From [the notes in my comment](https://github.com/runelite/runelite/pull/14828#issuecomment-1095741523), it would be better to do `source.getId() == NULL_8740` than to check the name.